### ISSUE TITLE
Azure Functions Runtime 2.0 breaking change release notice

### ISSUE THREAD

**Issue by fabiocav** [Azure/app-service-announcements-discussions#46]

**Comment by fabiocav** [comment]
The next release has the initial set of changes to implement on the plan outlined [here](https://github.com/Azure/azure-functions-host/wiki/Assembly-Resolution-in-Azure-Functions#how-will-this-be-improved). The article provides a high level overview of those changes, but I'll have more detailed information about the specific set of changes in this release and link to that from the release notes. I'll also add the link to the announcement when that information is available.

**Comment by milismsft** [comment]
@fabiocav I'm having problems after this update when using "EventHubOutput" via Java functions. I did "pin" the version to the one suggested and remove the .jar from the lib directory. The function seems to execute fine when I checked in the Azure Portal; however though the eventHub does not seem to receive any new events. Here's the function.json:
`{
  "scriptFile" : "../inventory-processor-1.0-SNAPSHOT.jar",
  "entryPoint" : "org.inventory.hub.POT.sell",
  "bindings" : [ {
    "type" : "timerTrigger",
    "name" : "timerInfo",
    "direction" : "in",
    "schedule" : "*/10 * * * * *"
  }, {
    "type" : "eventHub",
    "name" : "data",
    "direction" : "out",
    "connection" : "TRANSACTIONS_EVENT_HUB_CONNECTION_STRING",
    "eventHubName" : "eventhub-for-transactions"
  } ],
  "disabled" : false
}
`
Are these settings still the correct ones?

**Comment by milismsft** [comment]
It does not so far...

**Comment by davidebbo** [comment]
@milismsft please double check what version you are using from the FUnction app settings blade in portal. If it's correctly using the previous version, there should be no way that the new one is affecting you. It could very well be an unrelated issue.

**Comment by pragnagopa** [comment]
@milismsft - Can you share you App name here or indirectly following instructions here: [Sharing-Your-Function-App-name-privately](https://github.com/Azure/azure-functions-host/wiki/Sharing-Your-Function-App-name-privately). Also, please note that to use the latest runtime you need to wait for the maven tools update.

**Comment by milismsft** [comment]
I have working now; it looks like I still have to deploy the function app using a local copy of the Bin folder. When I tried last night I saw an error during the function execution complaining that some resource/dll was not found so I removed the step of copying and packaging the local Bin... 
Uff...  so many settings, so easy to get it wrong.

**Comment by jeromekol** [comment]
Hi, running into the issue that my function app cannot load the "System.Linq.Dynamic.Core" assembly:
"Could not load file or assembly 'System.Linq.Dynamic.Core, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null'. The system cannot find the file specified."

Reverted back to 2.0.11651-alpha so it works again. I'm using the latest Functions.SDK 1.0.13 and nuget package of System.Linq.Dynamic.Core (1.0.8.7). I'm only using Http Trigger functions.

**Comment by cyberprune** [comment]
We are getting this error:

Exception while executing function: compile-preview Could not load file or assembly 'System.CodeDom, Version=4.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'. The system cannot find the file specified. Could not load the specified file.

@jeromekol how did you revert to a specific runtime version (it works for us on 2.0.11651.0, but not 2.0.11776.0)

@milismsft how did you configure the local bin to copy or not?

**Comment by davidebbo** [comment]
@cyberprune please see https://github.com/Azure/app-service-announcements/issues/106 for instructions on reverting.

**Comment by cyberprune** [comment]
@davidebbo thanks, that's working for our old code now.

I am assuming that the resolution will happen automatically in the future, given you have uploaded the dll as part of the function package, but at the moment that's not the case until later releases of the runtime where this is fixed?

It's moving from a set of known versions, which you match in your nuget packages in the solution, to a solution where the functions runtime spawns a process, then loads dependent DLLs.

**Comment by davidebbo** [comment]
@fabiocav is the expert, and I will let him comment on what might be happening here.

**Comment by davidebbo** [comment]
@jonathanfishbein1 Thanks! In the meantime, feel free to revert to the previous version (per instructions in https://github.com/Azure/app-service-announcements/issues/106), whch should get you back up (please confirm that it does).

**Comment by jonathanfishbein1** [comment]
I can confirm that reverted does fix the issue

**Comment by jeromekol** [comment]
@davidebbo  2018-05-18T10:51:54.487, Id=1d908088-22c2-4db5-88e6-caa42b011adf

We are using the "Microsoft.Linq.Dynamic" Nuget package

**Comment by fabiocav** [comment]
@jeromekol the issue you're running into is similar to what is described above, impacted by the restriction on `System.*` assemblies

**Comment by robdmoore** [comment]
Hi,

This rollout has caused a break in my Azure Function.

I'm getting an exception that the cosmos connection string I'm injecting from config is empty.

The injection is here: https://github.com/dddwa/ddd-backend/blob/master/DDD.Functions/SessionizeReadModelSync.cs#L21 and the binding code is here: https://github.com/dddwa/ddd-backend/blob/master/DDD.Functions/Config/SessionizeReadModelSyncConfig.cs

Reverting `FUNCTIONS_EXTENSION_VERSION` to `2.0.11651-alpha` causes the break to be fixed.

LMK if you need any other info.

**Comment by mayoatbm** [comment]
Hey gents,

Our Azure Functions are suddenly returning json in pascal case  instead of the camel case they've been returning which breaks all our javascript front end applications.  

We haven't had any new deployments so the code is the same.  And I can confirm that reverting the runtime version to 2.0.11651-alpha fixes the problem.

Is this a planned change in case?

**Comment by rasmustherkelsen** [comment]
Same issue as mayoatbm reports. Breaks all my upstream logic apps.

**Comment by trydis** [comment]
Getting this on MacOS:

[22.05.2018 08:03:59] System.Private.CoreLib: Exception while executing function: SendActivity. Microsoft.Azure.WebJobs.Host: Exception binding parameter 'client'. Microsoft.Azure.DocumentDB.Core: The type initializer for 'Microsoft.Azure.Documents.UserAgentContainer' threw an exception. Microsoft.Azure.DocumentDB.Core: Error reading Darwin Kernel Version. Microsoft.Azure.WebJobs.Script: Object reference not set to an instance of an object.

**Comment by davidebbo** [comment]
Also listing various issues which may be related:

- https://github.com/Azure/Azure-Functions/issues/808
- https://github.com/Azure/Azure-Functions/issues/813
- https://social.msdn.microsoft.com/Forums/en-US/47c5ddfa-c592-4161-a980-eac5f08c3daf/binary-serialization-netstandard-20-tools-150405020

@fabiocav let's look through each reported issue and figure out whether it's caused by the `System.*` restriction or something else.

**Comment by fabiocav** [comment]
Reviewing all of those and posting updates.

**Comment by pragnagopa** [comment]
@fabiocav - Another related issue: https://github.com/Azure/azure-functions-host/issues/2881

**Comment by brandonh-msft** [comment]
@fabiocav 
w.r.t
> I'll have more detailed information about the specific set of changes in this release and link to that from the release notes

I see we're pushing this release now; can you elaborate on those changes here? Haven't seen anything deeper outlined anywhere.

**Comment by mayoatbm** [comment]
@fabiocav  We have six function apps, each with many http triggered functions (RESTful microservices). They all return json serialized from c# classes (serialized by the runtime/fx).  I'll look into the repro.

**Comment by fabiocav** [comment]
@rasmustherkelsen @mayoatbm there were some JSON serialization behavior changes caused by the assembly load work (expected), but that's not the case with your scenario. There was indeed a change to the default serialization behavior with this release, but that will be reverted. Apologies for the inconvenience.

**Comment by rasmustherkelsen** [comment]
@fabiocav thanks for the info. Is there a way to get a heads up before you deploy this fix so we can revert our compensating changes at about the same time?

**Comment by mayoatbm** [comment]
@fabiocav Thanks for the update. I have the same question as @rasmustherkelsen, we'd like to know when this is coming so we can adjust.

**Comment by brandonh-msft** [comment]
@rasmustherkelsen not for nothing, but one might argue that the compensations you put in place should stay there if you're fully dependent on the lower camelcase vs upper camelcase output ;)

**Comment by rasmustherkelsen** [comment]
@brandonh-msft you are right! That's how one would expect deserialisation to work... would you please forward that idea to the Logic Apps team as it is my function connector that breaks because it is case sensitive ;-)

**Comment by fabiocav** [comment]
For notifications, I would recommend subscribing to the [announcements repo](https://github.com/Azure/app-service-announcements/issues). We'll announce releases and point to release notes prior to deployment.
